webapp/src/ts/components/filters/facility-filter/facility-filter.component.ts

```diff
@@ -149,7 +160,13 @@ export class FacilityFilterComponent implements OnInit, AbstractFilter, AfterVie
 
   applyFilter(facilities) {
     const facilityIds = facilities.map(facility => facility.doc?._id);
-    this.globalActions.setFilter({ facilities: { selected: facilityIds } });
+    let selectedFacilities;
+
+    if (facilityIds.length) {
```

Comment:
What is the need for this if-check?  Is there a difference between `facilities` being `null` vs being an object where `selected` is an empty array?

Reply:
Same question applies to the `applyFilter` function for the other filters too.

Reply:
yes, there is an issue in master. Whenever the user searches, the system is recording [telemetry metrics](https://github.com/medic/cht-core/blob/5de528f9cb74dfff2522831d476081139bd0a09a/webapp/src/ts/services/search.service.ts#L146) about the filters (date, facility, etc) that have selected criteria.

If you select some places and forms, telemetry records `search:reports:places:forms`, then if you go and click "clear" from places. Then telemetry records `search:reports:places:forms`.
The expected record is `search:reports:forms`, because you're not using places filter anymore. 

We need these metrics to know what filters are being actively used by users.

Thinking that we are persisting the selection in rxjs store and that we aren't storing filters the same way (date is object, statuses is array, while forms and places are instance of Set, not sure about the original reason), then I thought it'd be better to store unused filters as `undefined`, to keep a standard filter state for empty, also to fix this issue.... and hopefully avoid other futures issues.

Reply:
It feels like a filter's responsibility to store it's data okay 🤔
Let me know your thoughts

Reply:
Not sure we are talking about the same thing.  Basically, was just wondering why this function needed to be refactored?  Since we are calling `setFilter` either way, it seems like we do not really gain anything from the extra logic.  But, it is not a big deal and should work fine how you have it.  :+1:

Reply:
Ahh yes, thank you for the detailed explanation. This makes sense now!